bundles/org.openhab.binding.panasonicbdp/src/main/java/org/openhab/binding/panasonicbdp/internal/handler/PanaBlurayHandler.java

```diff
@@ -0,0 +1,395 @@
+                    || CONTROL.equals(channelUID.getId())) {
+                final String commandStr;
+                if (command instanceof OnOffType) {
+                    commandStr = CMD_POWER + command; // e.g. POWERON or POWEROFF
+                    // if the power is switched on while the polling is running, the switch could bounce back to off,
+                    // set this flag to stop the first polling event from changing the state of the switch to give the
+                    // player time to start up and report the correct power status on the next poll
+                    debounce = false;
+                } else if (command instanceof PlayPauseType || command instanceof NextPreviousType
+                        || command instanceof RewindFastforwardType) {
+                    if (command == PlayPauseType.PLAY) {
+                        commandStr = CMD_PLAYBACK;
+                    } else if (command == PlayPauseType.PAUSE) {
+                        commandStr = CMD_PAUSE;
+                    } else if (command == NextPreviousType.NEXT) {
+                        commandStr = CMD_SKIPFWD;
+                    } else if (command == NextPreviousType.PREVIOUS) {
+                        commandStr = CMD_SKIPREV;
+                    } else if (command == RewindFastforwardType.FASTFORWARD) {
+                        commandStr = CMD_CUE;
+                    } else if (command == RewindFastforwardType.REWIND) {
+                        commandStr = CMD_REV;
+                    } else {
+                        logger.debug("Invalid control command: {}", command);
+                        return;
+                    }
+                } else {
+                    commandStr = command.toString();
+                }
+
+                // build the fields to POST the RC_ command string
+                Fields fields = new Fields();
+                fields.add("cCMD_RC_" + commandStr + ".x", "100");
+                fields.add("cCMD_RC_" + commandStr + ".y", "100");
+
+                // if command authentication enabled, get nonce to create authKey and add it to the POST fields
+                if (authEnabled) {
+                    if (playerKey.length() != 32) {
+                        logger.debug("Invalid playerKey length, was {}, should be 32", playerKey.length());
+                        updateStatus(ThingStatus.ONLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/error.keyerror");
+                        return;
+                    }
+
+                    final String nonce = sendCommand(GET_NONCE_CMD, nonceUrlStr).trim();
+                    if (nonce.isBlank()) {
+                        return;
+                    } else if (nonce.length() != 32) {
+                        logger.debug("Error retrieving nonce, message was: {}", nonce);
+                        updateStatus(ThingStatus.ONLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/error.nonce");
```

Comment:
```suggestion
                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/error.nonce");
```

Reply:
I did this so the status detail would stay 'sticky' with the communication error notification. If set to OFFLINE, it would immediately bounce back to ONLINE/none by lines 149-150 when the next polling run completes.

Reply:
The valid status combinations are here:
https://www.openhab.org/docs/concepts/things.html#status-details

What does it exactly mean when `nonce` doesn't have a length of 32? Just to understand what kind of situation this is.

Reply:
If the nonce response does not have length of 32, it is invalid. It probably won't happen but still seemed like a good sanity check before continuing.

Reply:
> The valid status combinations are here: https://www.openhab.org/docs/concepts/things.html#status-details
> 
I need to re-think how I am setting statuses. Based on the allowed statuses this will need to set OFFLINE/CONFIGURATION_ERROR and have it stick until the user makes a configuration change.

Reply:
For sure it is. So I guess this could be considered as something unexpected (e.g. a bug). So sinc you are already logging it, and assuming that the binding will quickly recover from this, you could simply choose to ignore it, i.e. not update status at all? Or you could keep `ThingStatus.ONLINE`, but change to `ThingStatusDetail.NONE` for compliance with valid combinations (but still provide the description).

Reply:
> I need to re-think how I am setting statuses. Based on the allowed statuses this will need to set OFFLINE/CONFIGURATION_ERROR and have it stick until the user makes a configuration change.

That would be appropriate indeed if this would be caused by invalid user configuration preferably detected during `initialize()`. When configuration is fixed, `initialize()` is called again, which should help resolving the issue.

Reply:
Ok, I moved one to initialize() and made the others Online/None with a description. Ready for another review when you have a moment.